rust-bindings/src/lib.rs

```diff
@@ -0,0 +1,194 @@
+            )
+        }
+    }
+
+    /// Matches any of a list of slices against the filter.
+    pub fn matches_any(&self, slices: &[&[u8]]) -> bool {
+        // Convert the slices to a type that C/C++ can understand.
+        let slices: Vec<bindings::Slice> = slices
+            .iter()
+            .map(|slice| bindings::Slice {
+                bytes: slice.as_ptr(),
+                length: slice.len(),
+            })
+            .collect();
+
+        // SAFETY: The length provided matches the number of slices, and the `GCSFilter`
+        // struct is guaranteed to be valid for the lifetime of the `Bip158Filter` struct, so this should be safe.
+        unsafe { bindings::filter_match_any(self.0, slices.as_ptr(), slices.len()) }
+    }
+}
+
+impl Drop for Bip158Filter {
+    fn drop(&mut self) {
+        // SAFETY: The `Slice` struct is guaranteed to be valid for the lifetime of the `Bip158Filter` struct.
+        unsafe { bindings::free_filter(self.0) }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashSet;
+
+    use rand::{RngCore, SeedableRng};
+    use rand_chacha::ChaCha8Rng;
+    use sha2::{digest::FixedOutput, Digest, Sha256};
+
+    use super::*;
+
+    fn hash(bytes: &[u8]) -> [u8; 32] {
+        let mut hasher = Sha256::new();
+        hasher.update(bytes);
+        hasher.finalize_fixed().into()
+    }
+
+    #[test]
+    fn test_filter() {
+        let elem1 = &hash(b"abc");
+        let elem2 = &hash(b"xyz");
+        let elem3 = &hash(b"123");
+        let not_elem1 = &hash(b"hello");
+        let not_elem2 = &hash(b"bye");
+
+        let filter = Bip158Filter::new(&[elem1, elem2, elem3]);
+
+        let encoded = filter.encode();
+        assert_eq!(encoded.as_ref(), [3, 174, 90, 204, 224, 219, 7, 253, 91]);
+        assert_eq!(&encoded, &encoded);
+
+        assert!(filter.matches(elem1));
+        assert!(filter.matches(elem2));
+        assert!(filter.matches(elem3));
+
+        assert!(!filter.matches(not_elem1));
+        assert!(!filter.matches(not_elem2));
+
+        assert!(filter.matches_any(&[elem1, elem2, elem3]));
+        assert!(filter.matches_any(&[elem1, elem2]));
+        assert!(filter.matches_any(&[elem3]));
+        assert!(filter
+            .matches_any(&[not_elem1, not_elem1, elem1, not_elem2, not_elem1, not_elem2, elem2]));
+
+        assert!(!filter.matches_any(&[]));
+        assert!(!filter.matches_any(&[not_elem1, not_elem2]));
+    }
+
+    #[test]
+    fn test_false_positive() {
+        let mut rng = ChaCha8Rng::seed_from_u64(0);
+
+        let hashes: Vec<[u8; 4]> = (0..100)
+            .map(|_| {
+                let mut hash = [0; 4];
+                rng.fill_bytes(&mut hash);
+                hash
+            })
+            .collect();
+
+        let mut hash_set = HashSet::new();
+        for hash in hashes.iter() {
+            hash_set.insert(hash);
+        }
+
+        let refs: Vec<&[u8]> = hashes.iter().map(|hash| hash.as_ref()).collect();
```

Comment:
I don't think you would need to allocate this vector just for the conversion into slices